authors/apps/profiles/views.py

```diff
@@ -0,0 +1,57 @@
+from rest_framework.generics import (
+    ListAPIView,
+    RetrieveUpdateAPIView)
+from rest_framework.permissions import (
+    IsAuthenticatedOrReadOnly,
+    AllowAny
+)
+from .renderers import ProfileJSONRenderer
+from .permissions import IsOwnerOrReadOnly
+from .serializers import ProfileSerializer
+from .models import Profile
+from .exceptions import ProfileDoesNotExist
+from rest_framework import status
+from rest_framework.response import Response
+
+
+class ListProfileView(ListAPIView):
+    permission_classes = (IsAuthenticatedOrReadOnly,)
+    queryset = Profile.objects.all()
+    serializer_class = ProfileSerializer
+    ordering_fields = ('timestamp')
+
+
+class RetrieveUpdateProfileView(RetrieveUpdateAPIView):
+    """
+    This method allows a user to view a profile
+    and only its owner to edit it
+    """
+    queryset = Profile.objects.all()
+    serializer_class = ProfileSerializer
+    permission_classes = (IsAuthenticatedOrReadOnly,
+                          IsOwnerOrReadOnly, AllowAny,)
+    renderer_classes = (ProfileJSONRenderer,)
+
+    def check_for_object(self, username):
+        try:
+            obj = Profile.objects.get(user__username=username)
+            return obj
+        except Profile.DoesNotExist:
+            raise ProfileDoesNotExist
+
+    def retrieve(self, request, username, *args, **kwargs):
```

Comment:
@NLSanyu You might not need to override the retrieve method if you override the get_object method as commented above. You may not need to re-write all this again because it is already done by the framework.

Reply:
@NLSanyu If you read through the resource I shared on the channel under any Retrieve generic class you get to the retrieve method is well implemented
`def retrieve(self, request, *args, **kwargs):
        instance = self.get_object()
        serializer = self.get_serializer(instance)
        return Response(serializer.data)`